tests/test_reporter.py

```diff
@@ -296,3 +260,40 @@ def send(_):
         yield reporter.close()
         assert reporter.queue.qsize() == 0, 'all spans drained'
         assert count[0] == 4, 'last span submitted in one extrac batch'
+
+    @gen_test
+    def test_composite_reporter(self):
```

Comment:
why does this function need to move?

Reply:
They made some changes in `Tornado 5+` (for asyncio compatibility) and done callback of future will fire on next loop cycle. That's why I made `yield f` to resolving future and it made the test asynchronous.